### PR TITLE
[#5278][Android] persist webview state during session

### PR DESCRIPTION
**Pull request by rasom**
implements (partially) #5278 
naїve implementation on java side https://github.com/status-im/react-native-webview-bridge/commit/c52d35c364040021591d0985249d1185b97ee1e8

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/browser/views.cljs [hunk 1]
```diff
@@ -100,7 +100,9 @@
                                 :handler   #(re-frame/dispatch [:navigate-to-modal :wallet-modal])}]]]
        [react/view components.styles/flex
         [components.webview-bridge/webview-bridge
-         {:ref                                   #(reset! webview %)
+         {:dapp?                                 dapp?
```
Comment (flexsurfer) [inline src/status_im/ui/screens/browser/views.cljs:104]: do we keep state only for dapps from our list?

Comment (rasom) [reply src/status_im/ui/screens/browser/views.cljs:104]: right now only for dapps, but we can use any other id instead of dapp-name

**Review by flexsurfer — APPROVED** [review]

**Review by jeluard — APPROVED** [review]

**Review by janherich — APPROVED** [review]

**Review by goranjovic — APPROVED** [review]

**Comment by Serhy** [comment]
Apparently, something is not right. Latest build checked on Android 6.0.1, 7.0 - no webview data persistance upon re-navigating to Dapp. Checked several Dapps and behaviour is the same to develop build. E.g. for `ERC dEX`
1) Open Status, create a new account and navigate to `Open Dapp` -> `dDEX`
2) Proceed to dapp home page
3) Set the values in 'Limit price' and 'Amount' to non-default (`0.0111` and `20` respectively)
4) Navigate back to Status Home screen
5) Tap `dDEX` from Home screen

Could it be something related to build config with CACHED_WEBVIEWS_ENABLED not set, @rasom?
TF session with log and video: https://app.testfairy.com/projects/4803622-status/builds/8545116/sessions/4397714732/?accessToken=ZTz7dgLVGOl38S-YtcCxUspNMhQ

<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803622-status/builds/8545116/sessions/4397714732/?accessToken=ZTz7dgLVGOl38S-YtcCxUspNMhQ">TestFairy: Status 0.9.25 (6295) @2018</a></strong></div></blockquote>

**Comment by rasom** [comment]
No, it's not related to CACHED_WEBVIEWS_ENABLED, works just fine on my real device, 6.0.1

**Comment by Serhy** [comment]
Latest build (Android) https://i.diawi.com/5eVvgQ (and iOS https://i.diawi.com/PNLjYV) looks good!
As it mentioned, webview on Android for Dapps in default list keeps the data when user navigates back to browser session. 
The #5278 still relevant for iOS, and for Dapps that not in Status Dapp list, but cool that we have fixed it for Android browser with Dapps in Status list. Thanks, @rasom !

Autotests reviewed and addressed for detail check (could not be reproduced manually). (by this moment test_send_stt_from_wallet , test_send_token_with_7_decimals, test_send_transaction_from_daap , test_send_eth_from_wallet_to_contact should be fixed. test_deploy_contract_from_daap will be fixed separately, all the rest failures are due to known existing GH issue)



<blockquote><div><strong><a href="https://i.diawi.com/5eVvgQ">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/PNLjYV">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Review by mandrigin — CHANGES_REQUESTED** [review]
since that also requires changes to the web bridge, let's wait for the image support first (it simplifies the logic)

**Comment by rasom** [comment]
> let's wait for the image support first

@mandrigin which PR is related to this change?

**Review by mandrigin — APPROVED** [review]

**Comment by rachelhamlin** [comment]
Hey @rasom this is great. Partially solved refers to Android only, right? Will you also work on iOS?

**Pull request merged**